src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -974,76 +867,92 @@ const put = async (frameworkName, config, rawConfig) => {
     // create priority class
-    await createPriorityClass(frameworkName, podPriority);
+    priorityClassDef = getPriorityClassDef(frameworkName, podPriority);
   }
 
   // send request to framework controller
   let response;
   try {
-    response = await k8sModel.getClient().request({
-      method: 'post',
-      url: launcherConfig.frameworksPath(),
-      headers: launcherConfig.requestHeaders,
-      data: frameworkDescription,
+    response = await axios({
+      method: 'put',
+      url: launcherConfig.writeMergerUrl + '/api/v1/frameworkRequest',
+      data: {
+        frameworkRequest: frameworkDescription,
+        submissionTime: submissionTime,
+        configSecretDef: configSecretDef,
+        priorityClassDef: priorityClassDef,
+        dockerSecretDef: dockerSecretDef,
+      },
+      headers: {
+        'Content-Type': 'application/json',
+      },
     });
   } catch (error) {
     if (error.response != null) {
       response = error.response;
     } else {
-      // do not await for delete
-      auths.length && deleteDockerSecret(frameworkName);
-      config.secrets && deleteJobConfigSecret(frameworkName);
-      launcherConfig.enabledPriorityClass && deletePriorityClass(frameworkName);
       throw error;
     }
   }
-  if (response.status !== status('Created')) {
-    // do not await for delete
-    auths.length && deleteDockerSecret(frameworkName);
-    config.secrets && deleteJobConfigSecret(frameworkName);
-    launcherConfig.enabledPriorityClass && deletePriorityClass(frameworkName);
+  if (response.status !== status('OK')) {
     throw createError(response.status, 'UnknownError', response.data.message);
   }
-  // do not await for patch
-  auths.length && patchDockerSecretOwner(frameworkName, response.data.metadata.uid);
-  config.secrets && patchJobConfigSecretOwner(frameworkName, response.data.metadata.uid);
 };
 
 const execute = async (frameworkName, executionType) => {
-  // send request to framework controller
   let response;
   try {
-    const headers = {...launcherConfig.requestHeaders};
-    headers['Content-Type'] = 'application/merge-patch+json';
-    response = await k8sModel.getClient().request({
-      method: 'patch',
-      url: launcherConfig.frameworkPath(encodeName(frameworkName)),
-      headers,
+    const framework = await databaseModel.Framework.findOne({
+      attributes: ['snapshot', 'submissionTime', 'configSecretDef', 'priorityClassDef', 'dockerSecretDef'],
+      where: {name: encodeName(frameworkName)},
+    });
+    if (!framework) {
+      throw createError('Not Found', 'NoJobError', `Job ${frameworkName} is not found.`);
+    }
+    const snapshot = JSON.parse(framework.snapshot);
+    const frameworkRequest = _.pick(snapshot, [
+      'apiVersion',
+      'kind',
+      'metadata.name',
+      'metadata.labels',
+      'metadata.annotations',
+      'spec',
+    ]);
+    frameworkRequest.spec.executionType = `${executionType.charAt(0)}${executionType.slice(1).toLowerCase()}`;
```

Comment:
No lock during this RMW, so update executionType may cause other's update during this period lost.

Can you move this RMW to write merge or let write merger support things like k8s patch?

Reply:
Currently we only modify one field `executionType`. So the lost-update problem won't happen.

I suggest to add a patch api in write merger in future.

Reply:
Better to solve it in current PR, since we will also update tasknumber, completionpolicy etc.
If it is too hard, pls explain the reason, comment it and track in issue

Reply:
It is not hard. I will implement a patch interface using `json merge patch`.

Reply:
Great. Given we only have a few update scenarios, your implementation does not need too complex or too general